app/for_act/defeat.py

```diff
@@ -0,0 +1,4 @@
+def defeat_check(knight_list: list) -> dict:
+    for knight in knight_list:
+        if knight["hp"] <= 0:
+            knight["hp"] = 0

```

Comment:
It doesn't return dict. The same in other cases. But you can add an annotation that `list[dict]`